src/support/rand.c

```diff
@@ -169,7 +169,7 @@ __wt_random(WT_RAND_STATE volatile *rnd_state) WT_GCC_FUNC_ATTRIBUTE((visibility
 
     M_W(rnd) = w = 18000 * (w & 65535) + (w >> 16);
     M_Z(rnd) = z = 36969 * (z & 65535) + (z >> 16);
-    *rnd_state = rnd;
+    WT_WRITE_ONCE(*rnd_state, rnd);
```

Comment:
I've removed all other instances of WT_WRITE_ONCE from `rand.c` but kept this one as it still aligns with the above ORDERED_READ. I have also converted that over to a WT_READ_ONCE.